### PR TITLE
Remove defaults and add manual flag parsing for configuration

### PR DESCRIPTION
**Pull request by keremgocen**
Introduce manual parsing for CLI flags related to Neo4j configuration, allowing users to override environment variables. Enhance test coverage for edge cases and integration tests with real environment variables.

### CODE REVIEW

File: internal/cli/args.go [hunk 1]
```diff
@@ -14,38 +15,101 @@ Usage:
...
 
 Options:
-  -h, --help      Show this help message
-  -v, --version   Show version information
+  -h, --help                          Show this help message
+  -v, --version                       Show version information
+  --neo4j-uri <URI>                   Neo4j connection URI (overrides env var)
+  --neo4j-username <USERNAME>         Database username (overrides env var)
+  --neo4j-password <PASSWORD>         Database password (overrides env var)
+  --neo4j-database <DATABASE>         Database name (overrides env var)
```
Comment (mjfwebb) [inline internal/cli/args.go]: I think I would like to be a bit more explicit here

```suggestion
  --neo4j-uri <URI>                   Neo4j connection URI (overrides environment variable NEO4J_URI)
  --neo4j-username <USERNAME>         Database username (overrides environment variable NEO4J_USERNAME)
  --neo4j-password <PASSWORD>         Database password (overrides environment variable NEO4J_PASSWORD
  --neo4j-database <DATABASE>         Database name (overrides environment variable NEO4J_DATABASE)

```

File: internal/cli/args.go [hunk 2]
```diff
@@ -14,38 +15,101 @@ Usage:
...
-  NEO4J_PASSWORD  Database password (required)
+Required Environment Variables:
+  NEO4J_URI       Neo4j database URI
+  NEO4J_USERNAME  Database username
+  NEO4J_PASSWORD  Database password
+
+Optional Environment Variables:
   NEO4J_DATABASE  Database name (default: neo4j)
+  NEO4J_TELEMETRY Enable/disable telemetry (default: true)
+  NEO4J_READ_ONLY Enable read-only mode (default: false)
```
Comment (mjfwebb) [inline internal/cli/args.go:36]: We don't want these to be args?

File: internal/cli/args.go [hunk 3]
```diff
@@ -14,38 +15,101 @@ Usage:
...
+- os.Args[4] = "test"
+
+As the loop processes:
+1. i=1: Matches case "--neo4j-uri" → i += 2 → i=3 (skips the URI value)
+2. i=3: Matches case "--neo4j-username" → i += 2 → i=5 (skips the "test" value)
+3. i=5: Loop ends
+
+This allows the arguments to "pass through" untouched so that flag.Parse() in main.go can later handle them properly.
+*/
+
```
Comment (mjfwebb) [inline internal/cli/args.go]: Maybe it's just me but I find this unnecessarily verbose, when the code itself also explains how this works.

File: cmd/neo4j-mcp/main.go [hunk 4]
```diff
@@ -19,12 +20,37 @@ var MixPanelEndpoint = ""
 var MixPanelToken = ""
 
 func main() {
+	// Define CLI flags for configuration
```
Comment (mjfwebb) [inline cmd/neo4j-mcp/main.go]: Just a personal choice here, but I would have likely put all of these code additions into `config.LoadConfig()` instead of polluting `main()`

**Review by mjfwebb — COMMENTED** [review]
Very nice! Left some comments for your consideration 👍

File: cmd/neo4j-mcp/main.go (hunk 4 above)
Comment (MacondoExpress) [reply cmd/neo4j-mcp/main.go]: +1

File: cmd/neo4j-mcp/main.go [hunk 5]
```diff
@@ -2,6 +2,7 @@ package main
 
 import (
 	"context"
+	"flag"
```
Comment (MacondoExpress) [inline cmd/neo4j-mcp/main.go]: what a nice std library

File: cmd/neo4j-mcp/main.go [hunk 6]
```diff
@@ -19,12 +20,37 @@ var MixPanelEndpoint = ""
...
-	cfg, err := config.LoadConfig()
-	if err != nil {
+	// Parse flags after HandleArgs
+	flag.Parse()
+
+	// Load config from environment variables
+	cfg := config.LoadConfig()
+
+	// Override config with CLI flags if provided (CLI flags take precedence)
+	if *neo4jURI != "" {
```
Comment (MacondoExpress) [inline cmd/neo4j-mcp/main.go]: Same as @mjfwebb pointed out, I would delegate this "overwrite" to the config package, same as "cfg.Validate()"  I see no benefits to delegate the consumer to correctly invoke it, I would delegate that method to be consumed by `config.loadConfig`

File: cmd/neo4j-mcp/main.go (hunk 4 above)
Comment (MacondoExpress) [reply cmd/neo4j-mcp/main.go]: IMO, here we should delegate the "cli" and "config" with clear separation of ownership, from the main.go file I would keep the consumer logic as dummy as possible:
- "handleArgs" we could return a typed struct with the expected args so that can be used later by "loadConfig"
- "loadConfig" that returns errors in case of validation errors

File: test/integration/config/config_test.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+// TestLoadConfig_Neo4jConnectivity tests that the application can connect to Neo4j with the provided configuration.
+// This test requires the following environment variables to be set:
+//   - NEO4J_URI: Neo4j database URI
+//   - NEO4J_USERNAME: Database username
+//   - NEO4J_PASSWORD: Database password
+//
+// If any of these environment variables are missing, the test will fail with a clear error message.
+// If the environment variables are set but the database is not accessible, the test will fail with a connection error.
+func TestLoadConfig_Neo4jConnectivity(t *testing.T) {
```
Comment (MacondoExpress) [inline test/integration/config/config_test.go]: Let's remove this test, as we can test entirely with unit testing, without undesired side effects!

File: internal/cli/args.go (hunk 2 above)
Comment (keremgocen) [reply internal/cli/args.go:36]: converting them to args as well, I think it makes sense

File: internal/cli/args.go (hunk 1 above)
Comment (keremgocen) [reply internal/cli/args.go]: done

File: cmd/neo4j-mcp/main.go (hunk 6 above)
Comment (keremgocen) [reply cmd/neo4j-mcp/main.go]: moved the flag parsing logic into LoadConfig

File: cmd/neo4j-mcp/main.go (hunk 4 above)
Comment (keremgocen) [reply cmd/neo4j-mcp/main.go]: This should be done now, handleArgs is returning a struct to override the env vars from the cli flag parsing
loadConfig returns errors if validation fails

File: cmd/neo4j-mcp/main.go [hunk 8]
```diff
@@ -50,10 +60,10 @@ func main() {
 
 	anService := analytics.NewAnalytics(MixPanelToken, MixPanelEndpoint, cfg.URI)
 
-	if cfg.Telemetry == "false" || MixPanelEndpoint == "" || MixPanelToken == "" {
+	if !cfg.Telemetry || MixPanelEndpoint == "" || MixPanelToken == "" {
 		log.Println("Telemetry disabled.")
 		anService.Disable()
-	} else if cfg.Telemetry == "true" {
+	} else {
 		anService.Enable()
```
[the remark this answers is not in the document]
Comment (keremgocen) [reply cmd/neo4j-mcp/main.go:75]: reverted the logic so that we only enable telemetry when all conditions are met at the same time

File: internal/cli/args.go [hunk 9]
```diff
@@ -14,38 +16,116 @@ Usage:
...
+				err = fmt.Errorf("%s requires a value (got flag %s instead)", arg, nextArg)
+				break
+			}
+			// Safe to skip flag and value - let flag package handle them
+			i += 2
 		default:
-			err = fmt.Errorf("unknown flag or argument: %s", arg)
+			if arg == "--" {
+				// Stop processing our flags, let flag package handle the rest
+				i = len(os.Args)
```
[the remark this answers is not in the document]
Comment (keremgocen) [reply internal/cli/args.go:120]: It doesn't work correctly in this context (it only breaks out of the switch, not the loop). 
We have two valid options:
1. Keep `i = len(os.Args)` (current) - Works correctly, though less conventional
2. Use labelled break - More idiomatic Go, but adds complexity

I think I'd like to stick to the `i = len(os.Args)` instead of a labelled break, which ends the loop and the code works correctly.

[the remark this answers is not in the document]
Comment (mjfwebb) [reply internal/cli/args.go:120]: No strong opinions here. I think either way is fine, though I could make the argument that a labelled break is quicker to understand.

File: internal/cli/args.go [hunk 10]
```diff
@@ -14,38 +16,116 @@ Usage:
...
+		// Allow configuration flags to be parsed by the flag package
+		case "--neo4j-uri", "--neo4j-username", "--neo4j-password", "--neo4j-database", "--neo4j-read-only", "--neo4j-telemetry":
+			// Check if there's a value following the flag
+			if i+1 >= len(os.Args) {
+				err = fmt.Errorf("%s requires a value", arg)
+				break
+			}
+			// Check if next argument is another flag (starts with --)
+			nextArg := os.Args[i+1]
+			if strings.HasPrefix(nextArg, "-") {
```
[the remark this answers is not in the document]
Comment (keremgocen) [reply internal/cli/args.go:111]: 😭 you just suggested the `--` earlier:)

File: internal/cli/args.go [hunk 11]
```diff
@@ -14,38 +16,116 @@ Usage:
...
+			i++
 		case "-v", "--version":
 			flags["version"] = true
+			i++
+		// Allow configuration flags to be parsed by the flag package
+		case "--neo4j-uri", "--neo4j-username", "--neo4j-password", "--neo4j-database", "--neo4j-read-only", "--neo4j-telemetry":
+			// Check if there's a value following the flag
+			if i+1 >= len(os.Args) {
+				err = fmt.Errorf("%s requires a value", arg)
+				break
```
[the remark this answers is not in the document]
Comment (keremgocen) [reply internal/cli/args.go:107]: we exit the loop on #L127 if there's an error in the switch so that's incorrect

File: internal/cli/args.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (keremgocen) [reply internal/cli/args.go:120]: We decided to revisit this at a later stage and think about whether we want to use some helper packages similar to in aura-cli for the flags.
